vllm_detector_adapter/protocol.py

```diff
@@ -173,6 +173,42 @@ class ContextAnalysisRequest(BaseModel):
     # is identified, it can be implemented here.
 
 
+##### Generation Detection types for the /text/generation detection endpoint
+
+
+class GenerationDetectionRequest(BaseModel):
+    # Prompt to a model
+    prompt: str = Field(examples=["What is a moose?"])
+    # Generated text from a model
+    generated_text: str = Field(examples=["A moose is not a goose."])
+    # Parameters to pass through to chat completions, optional
+    detector_params: Optional[Dict] = {}
+
+    def to_chat_completion_request(self, model_name: str):
+        """Function to convert [fms] generation detection request to openai chat completion request"""
+        messages = [
+            {"role": "user", "content": self.prompt},
+            {"role": "assistant", "content": self.generated_text},
```

Comment:
this works for now, but might need to change with the function hallucination type of features 🤔